File: contributions/jocamlJoinPatterns/meganalysis.ml
open Company;;

(* sample company meganalysis *)
let craig = employee "Craig" "Redmond" 123456.;;
let erik = employee "Erik" "Utrecht" 12345.;;
let ralf = employee "Ralf" "Koblenz" 1234.;;
let ray = employee "Ray" "Redmond" 234567.;;
let klaus = employee "Klaus" "Boston" 23456.;;
let karl = employee "Karl" "Riga" 2345.;;
let joe = employee "Joe" "Wifi City" 2344.;;

let research = department "Research" craig [erik; ralf] [];;
let dev11 = department "Dev11" karl [joe] [];;
let dev1 = department "Dev1" klaus [] [dev11];;
let development = department "Development" ray [] [dev1];;

let meganalysis = company "meganalysis" [research; development];;

File: contributions/clojure/src/org/softlang/meganalysis.clj
(ns org.softlang.meganalysis
  (:require org.softlang.company)
  (:use org.softlang.scenarios)
  (:import (org.softlang.company Dept Company Employee)))

;; sample company

(def meganalysis (Company. "meganalysis"
                   [(Dept. "Research"
                      (Employee. "Craig" "Redmond" 123456)
	                      [(Employee. "Erik" "Utrecht" 12345) 
	                       (Employee. "Ralf" "Koblenz" 1234)]
                      [])
                    (Dept. "Development"
                      (Employee. "Ray" "Redmond" 234567)
                      []
                      [(Dept. "Dev1"
                        (Employee. "Klaus" "Boston" 23456)
                        []
                        [(Dept. "Dev11"
                           (Employee. "Karl" "Riga" 2345)
                           [(Employee. "Joe" "Wifi City" 2344)]
                           [])])])]))
